client/src/main/scala/org/zalando/nakadi/client/Client.scala

```diff
@@ -0,0 +1,218 @@
+package org.zalando.nakadi.client
+
+import scala.concurrent.Future
+import akka.actor.Terminated
+import akka.http.scaladsl.model.HttpResponse
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.ExecutionContext.Implicits.global
+import org.slf4j.LoggerFactory
+import com.typesafe.scalalogging.Logger
+import akka.http.scaladsl.unmarshalling.Unmarshaller
+import akka.protobuf.ByteString
+import org.zalando.nakadi.client.model.EventType
+import akka.http.scaladsl.unmarshalling._
+import org.zalando.nakadi.client.model.DefaultMarshaller
+
+trait Client {
+
+  /**
+   * Retrieve monitoring metrics
+   *
+   * {{{
+   * curl --request GET /metrics
+   * }}}
+   */
+  def metrics(): Future[HttpResponse]
+
+  /**
+   * Returns a list of all registered EventTypes.
+   *
+   * {{{
+   * curl --request GET /event-types
+   * }}}
+   *
+   */
+  def eventTypes()(implicit marshaller: FromEntityUnmarshaller[EventType], charset: String = "UTF-8"): Future[EventType]
+
+  /**
+   * Creates a new EventType.
+   *
+   * {{{
+   * curl --request POST -d @fileWithEventType /event-types
+   * }}}
+   *
+   * @param event - The EventType to create.
+   *
+   */
+  def newEventType(eventType: Any): Future[Boolean]
```

Comment:
IMO noun based method does fit for write operations but it makes totally sense for read operations. Otherwise, it is quite confusing (at least for me)